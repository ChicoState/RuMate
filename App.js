import React from 'react';
import firebase from 'firebase';
import { StyleSheet } from 'react-native';
import { useScreens } from 'react-native-screens';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { createBottomTabNavigator } from 'react-navigation-tabs';
import { createStackNavigator } from 'react-navigation-stack';
import Icon from 'react-native-vector-icons/FontAwesome';
/* our imports */
import HomeScreen from './src/screens/HomeScreen';
import TasksScreen from './src/screens/TasksScreen';
import AuthScreen from './src/screens/AuthScreen';
import CreateTaskScreen from './src/screens/CreateTaskScreen';
import AddRoommateScreen from './src/screens/AddRoomateScreen';
import CalendarScreen from './src/screens/CalendarScreen';
import BillsScreen from './src/screens/BillsScreen';
import CreateBillScreen from './src/screens/CreateBillScreen';
import CreateRoommateGroupScreen from './src/screens/CreateRoommateGroupScreen';
import MessageScreen from './src/screens/MessagesScreen';
import CreateConversationScreen from './src/screens/CreateConversationScreen';
import ConversationScreen from './src/screens/ConversationScreen';
import InvitationsScreen from './src/screens/InvitationsScreen';
import AccountScreen from './src/screens/AccountScreen';
import PriorityScreen from './src/screens/PriorityScreen';
import ChangeDetailsScreen from './src/screens/ChangeDetailsScreen';

useScreens();
// ################################ //
// Remove to debug errors/warnings  //
   console.disableYellowBox = true; //
// ################################ //
const firebaseConfig = {
  apiKey: "AIzaSyA_ZWMR-MFG_ZHNK4_WuEeHoLP9vzsY_Vk",
  authDomain: "rumate-faaeb.firebaseapp.com",
  databaseURL: "https://rumate-faaeb.firebaseio.com",
  projectId: "rumate-faaeb",
  storageBucket: "",
  messagingSenderId: "484837108351",
  appId: "1:484837108351:web:53a875e9546d00aa2dad4a"
};
// Initialize Firebase
firebase.initializeApp(firebaseConfig);

const tabNavigator = createBottomTabNavigator({
  Home: {
    screen: HomeScreen,
    navigationOptions: {
      tabBarLabel: "Home",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="home" size={20} />
      )
    },
  },
  Tasks: {
    screen: TasksScreen,
    navigationOptions: {
      tabBarLabel: "Tasks",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="tasks" size={20} />
      )
    },
  },
  Bills: {
    screen: BillsScreen,
    navigationOptions: {
      tabBarLabel: "Bills",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="dollar" size={20} />
      )
    },
  },
  Messages: {
    screen: MessageScreen,
    navigationOptions: {
      tabBarLabel: "Messages",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="comments" size={20} />
      )
    },
  }
},
{
  initialRouteName: 'Home',
});

const taskNavigator = createStackNavigator({
  CreateTask: CreateTaskScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const appStackNavigator = createStackNavigator({
  Priority: PriorityScreen,
  Invitations: InvitationsScreen,
  CreateRoommateGroup: CreateRoommateGroupScreen,
  AddRoommate: AddRoommateScreen,
  Home: tabNavigator,
  Tasks: taskNavigator,
  CreateBill: CreateBillScreen,
  TaskCalendar: CalendarScreen,
  Messages: MessageScreen,
  CreateConversation: CreateConversationScreen,
  Conversation: ConversationScreen,
  Account: AccountScreen,
  ChangeDetails: ChangeDetailsScreen

},{
  initialRouteName: 'Home',
  defaultNavigationOptions: {
    header: null
  }
});

const authStack = createStackNavigator({
  Login: AuthScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const switchNav = createSwitchNavigator({
  App: appStackNavigator,
  Auth: authStack,
}, {
  initialRouteName: 'Auth',
  defaultNavigationOptions: {
    header: null
  }
});

const App = createAppContainer(switchNav);

const styles = StyleSheet.create({
  headerLeft: {
    fontSize: 20
  }
});

export default () => {
  return (
    <App />
  );
}
