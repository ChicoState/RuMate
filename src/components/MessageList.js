import React, { useState, useEffect } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import Message from '../components/Message';
import { ScrollView, FlatList } from 'react-native-gesture-handler';
import firebase from 'firebase';

const MessageList = ({ id, recipient }) => {
  const [messages, setMessages] = useState([]);
  const getMessages = () => {
    setMessages([])
    let response = firebase.database().ref('/messages');
    let users = firebase.database().ref('/users');
    let sender = "";
    users.on("value", (snapshot) => {
      let data = snapshot.val();
      for (let item in data) {
        if (firebase.auth().currentUser.uid == data[item].uid) {
          sender = data[item].name;
        }
      }
    })
    let messageList = [];
    response.on("value", (snapshot) => {
      let data = snapshot.val();
      for (let item in data) {
        if ((data[item].to == recipient && data[item].from == sender) ||
            (data[item].to == sender && data[item].from == recipient)) {
          console.log(data[item])
          messageList.push(data[item])
          setMessages(messageList);
        }
      }
    })
  }

  const renderMessages = () => {
    return (
      <FlatList style={styles.container}
        data={messages}
        keyExtractor={item => item.msgID}
        inverted={0}
        renderItem={(item) => {
          let sentToMe = false;
          if (item.item.senderID == firebase.auth().currentUser.uid) {
            sentToMe = true;
          }
          return (
            <Message 
              sentToMe={sentToMe}
              msg={item.item.text}
              time={item.item.time}
            />
          )
        }}
      />
    )
  }

  useEffect(() => {
    getMessages();
  }, []);

  return renderMessages();
}

const styles = StyleSheet.create({
  container: {
    height: "70%"
  }
});

export default MessageList;