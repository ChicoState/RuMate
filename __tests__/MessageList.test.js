import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App'
import MessageList from '../src/components/MessageList'

describe('<MessageList />', () => {
  it('has 1 child', () => {
    <App />
    const tree = renderer.create(<MessageList />).toJSON()
    expect(tree.children.length).toBe(1)
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<MessageList />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});