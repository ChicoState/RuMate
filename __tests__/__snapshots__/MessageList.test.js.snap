// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageList /> renders correctly 1`] = `
<RCTScrollView
  collapsable={false}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  invertStickyHeaders={1}
  inverted={1}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "transform": Array [
          Object {
            "scaleY": -1,
          },
        ],
      },
      Object {
        "height": "65%",
      },
    ]
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
